### ISSUE TITLE
File change inside container will be reset after host reboot.

### ISSUE THREAD

**Issue by ondh** [nestybox/sysbox#769]
Hi.

My problem is all file change inside container will be reset after host reboot.

this is my test step:

## Prepare test container

- install Ubuntu 22.04 server (kernel: Linux server 5.15.0-83-generic)
- install docker user get-docker.sh
- install sysbox use release deb file
- change docker config `/var/lib/docker` to `/data/lib/docker`
- change sysbox-mgr run data-dir to `/data/lib/sysbox` by add `--data-root` to /lib/systemd/system/sysbox-mgr.service
- systemctl daemon-reload and reboot.

- run test container: `docker run -itd --name=test --runtime=sysbox-runc nestybox/ubuntu-jammy-systemd-docker`
- make test file: `docker exec -it test touch /test.file`
- show file: `docker exec -it test ls /test.file`

```sh
root@server:~# docker exec -it test ls /test.file   
/test.file
```

### Test 1

```sh
root@server:~# docker exec -it test touch /test.file
root@server:~# docker exec -it test ls /test.file
/test.file
root@server:~# reboot now
# after reboot
root@server:~# docker start test
root@server:~# docker exec -it test ls /test.file   
ls: cannot access '/test.file': No such file or directory
```

### Test 2

```sh
root@server:~# docker exec -it test touch /test.file
root@server:~# docker exec -it test ls /test.file
/test.file
root@server:~# systemctl restart docker
# restart docker without restart sysbox. container is stopped
root@server:~# docker start test
Error response from daemon: failed to create shim task: OCI runtime create failed: container_linux.go:300: failed to chown rootfs clone caused: failed to invoke ChownClonedRootfs via grpc: rpc error: code = Unknown desc = invalid argument: unknown
Error: failed to start containers: test
```

### Test 3

```sh
root@server:~# docker exec -it test touch /test.file
root@server:~# docker exec -it test ls /test.file
/test.file
root@server:~# systemctl restart sysbox
# restart sysbox without restart docker. container is running
root@server:~# docker exec -it test ls /test.file
OCI runtime exec failed: exec failed: container_linux.go:424: starting container process caused: process_linux.go:102: starting setns process caused: fork/exec /proc/self/exe: no such file or directory: unknown
```

### Test 4

```sh
root@server:~# docker exec -it test touch /test.file
root@server:~# docker exec -it test ls /test.file
/test.file
root@server:~# systemctl restart sysbox
root@server:~# systemctl restart docker
# restart docker after restart sysbox. same as reboot server
root@server:~# docker start test
root@server:~# docker exec -it test ls /test.file
ls: cannot access '/test.file': No such file or directory
```

### Test 5

```sh
root@server:~# docker exec -it test touch /test.file
root@server:~# docker exec -it test ls /test.file
/test.file
root@server:~# systemctl restart docker
root@server:~# systemctl restart sysbox
# restart sysbox after restart docker. same as reboot server and Test 4
root@server:~# docker start test
root@server:~# docker exec -it test ls /test.file
ls: cannot access '/test.file': No such file or directory
```

**Comment by ctalledo** [comment]
Hi @ondh, thanks for giving Sysbox a try and for reporting the issue.

I think there's definitely a bug in Sysbox, related to keeping the state of the container across host reboots (or even sysbox restarts). 

I investigated, and I am pretty sure the bug occurs when the host machine either

* Does not have the [shiftfs](https://github.com/nestybox/sysbox/blob/master/docs/user-guide/design.md#shiftfs-module) module.
* Or has kernel < 5.19.

I am pretty sure that's the scenario you have. In this scenario, Sysbox uses a technique called container "rootfs-cloning", and the bug is that Sysbox is not keeping that rootfs clone state around when Sysbox is restarted. 

As a work-around, if you install shiftfs on the host, or alternatively upgrade the kernel to 5.19+, you will not see the issue. I tested on my Ubuntu Jammy host with kernel 5.19+ and was not able to repro.
